src/editors/sharedComponents/InsertLinkModal/api.js

```diff
@@ -0,0 +1,29 @@
+/* eslint-disable import/prefer-default-export */
+import { camelCaseObject, getConfig } from '@edx/frontend-platform';
+import { getAuthenticatedHttpClient } from '@edx/frontend-platform/auth';
+
+export const getBlocksFromCourse = async (courseId) => {
+  try {
+    const courseIdFormat = encodeURIComponent(courseId);
+    const { data } = await getAuthenticatedHttpClient().get(
+      `${
+        getConfig().LMS_BASE_URL
+      }/api/courses/v1/blocks/?course_id=${courseIdFormat}&all_blocks=true&depth=all&requested_fields=name,parent, display_name,block_type,children`,
+    );
+
+    const { blocks } = data;
+    const blocksFormat = Object.keys(blocks).reduce(
+      (prevBlocks, key) => ({
+        ...prevBlocks,
+        [key]: camelCaseObject(blocks[key]),
+      }),
+      {},
+    );
```

Comment:
All object in blocks response come in `snake_case` this converts all of them to `camelCase` for avoiding problems in when we need this data